snapshot/git/gitdb/bundlestore.go

```diff
@@ -225,27 +225,10 @@ func (s *bundlestoreSnapshot) Download(db *DB) error {
 		return err
 	}
 
-	// we are missing prereqs, so let's try updating the stream that's the basis of the bundle
-	// this likely happened because:
-	// we're in a worker that started at time T1, when master pointed at commit C1
-	// at time T2, a commit C2 was created in our stream
-	// at time T3, a user ingested a git commit C3 whose ancestor is C2
-	// GitDB in their scoot-snapshot-db picked a merge-base of C2, because T3-T2 was sufficiently
-	// large (say, a half hour) that it's reasonable to assume its easy to get.
-	// Now we've got the bundle for C3, which depends on C2, but we only have C1, so we have to
-	// update our stream.
-	if err := db.stream.updateStream(s.streamName, db); err != nil {
-		log.Infof("Couldn't download sha: %s, updateStream returned error: %s", s.SHA(), err.Error())
-		return err
-	}
-
-	if _, err := db.dataRepo.Run("bundle", "unbundle", filename); err != nil {
-		// if we still can't unbundle, then the bundle might be corrupt or the
-		// prereqs might not be in the stream, or maybe the git server is serving us
-		// stale data.
-		log.Infof("Couldn't download sha: %s, the final unbundling attempt returned error: %s", s.SHA(), err.Error())
-		return err
-	}
+	// NOTE updating from a stream for bundlestore snapshots is no longer supported
```

Comment:
what if instead of removing this functionality from OSS, we just used a different snapshot-satisfying interface internally with a Download() that returns error? In the general case (non-journaled repos), fetching to update bundles still seems useful.

Reply:
I'm against duplicating slightly altered functionality internally.

I was considering making this behavior configurable, and somehow marking it as disabled when setting up the internal db - this would be pretty easy to do without disabling anything, WDYT?

Longer term, I think this is something we should consider removing entirely, and instead of hiding this behavior in "gitdb" snapshot semantics, greatly simplify the snapshot operations, and delegate all update/stream behavior to filer updaters - a checkout failure in that case would fall back to an explicit filer update attempt.

Reply:
I'm against the duplication too - I was thinking more along the lines of something like making the git.DB's stream field an interface, and creating a noFetchStreamBackend internally that errors on updateStream.

Making it configurable could end up being less involved than that though, so I'm fine with that.

Agreed re: filer changes.